Use new quaternion.canonical(), reuse matrix::sign()

MaEtUgR opened the pull request:
**Describe problem solved by this pull request**
As a follow up to https://github.com/PX4/Matrix/pull/116/
- I want to take advantage of the very useful `canonical()` method that @kamilritz added for quaternions instead of implementing different variants in multiple places.
- I make sure the `sign()` function that I originally added to the mathlib is not a duplicate because it got copied for use in the matrix library here https://github.com/PX4/Matrix/pull/116/files#diff-54949d0fd557988a948ea903fee99e69R81. We can just use the one version that gets included.

**Test data / coverage**
- I added arbitrary antipodal quaternions in the AttitudeControl unit tests and made sure the tests fail without the "canonicalization" and pass with.
- I made sure the sign function is to the character the exact same.

MaEtUgR commented:
CI shows there's a problem to compile a matrix function for the IO.

MaEtUgR commented:
It requires https://github.com/PX4/Matrix/pull/125

kamilritz commented on src/lib/flight_tasks/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp:
```diff
@@ -301,7 +301,7 @@ void FlightTaskAutoLineSmoothVel::_prepareSetpoints()
 		else if (z_pos_setpoint_valid) {
 			// Use Z position setpoint to generate a Z velocity setpoint
 
```
We using here matrix::sign and above only sign()

MaEtUgR commented:
I
- rebased on master
- fixed the additional `sign()` that was used in the hover thrust estimator improvements
- pulled in my matrix library changes that fix the `stdout` dependency caused by a print test

jkflying commented on src/lib/mathlib/math/Functions.hpp:
```diff
@@ -41,15 +41,11 @@
-namespace math
-{
+#include <matrix/matrix/math.hpp>
+using matrix::sign;
```
I'm a little hesitant to put `sign` into the global namespace, since it may collide with a C function in a library an there is no way to fix it on the C side.

MaEtUgR replied:
Oh you're right that obviously gets through the include chain. Let me fix that.

MaEtUgR replied:
Fixed, thanks for catching.

MaEtUgR commented:
I added a non-functional `using namespace matrix;` instead of using all individual things. See https://github.com/PX4/Firmware/compare/fcfddd2d5504484b75ae5337ac59c7103269690f..d963bdafd6de910bdcf17ed259191c93ac46fc91
Will merge afterward.

MaEtUgR commented:
My only change since it passed described in my last comment cannot possibly cause
```
[2020-03-17T08:36:17.520Z] ======================================================================
[2020-03-17T08:36:17.520Z] FAIL: test_microbench_uorb (__main__.TestHadrwareMethods)
[2020-03-17T08:36:17.520Z] ----------------------------------------------------------------------
[2020-03-17T08:36:17.520Z] Traceback (most recent call last):
[2020-03-17T08:36:17.520Z]   File "./Tools/HIL/run_tests.py", line 128, in test_microbench_uorb
[2020-03-17T08:36:17.520Z]     self.assertTrue(do_test(self.TEST_DEVICE, self.TEST_BAUDRATE, "microbench_uorb"))
[2020-03-17T08:36:17.520Z] AssertionError: False is not true
[2020-03-17T08:36:17.520Z] 
[2020-03-17T08:36:17.520Z] ----------------------------------------------------------------------
```
Sorry but I'll ignore it.
